<?xml version="1.0"?>
<robot name="scooba" xmlns:xacro="http://www.ros.org/wiki/xacro"> 
  <!-- Constants for robot dimensions -->
  <xacro:property name="PI" value="3.1415926535897931"/>

  <!-- Import Rviz colors -->
  <xacro:include filename="$(find scooba_base)/urdf/materials.xacro" />

  <!-- Base Link -->
  <link name="base_link">
    <visual name="base">
      <origin xyz="0 0 0.045" rpy="0 0 0"/>
      <geometry>
	    <cylinder length="0.07" radius="0.18"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <visual name="caster">
      <origin xyz="0.16 0.0 0.01" rpy="0 0 0"/>
      <geometry>
        <sphere  radius="0.01"/>
      </geometry>
      <material name="grey"/>
    </visual>
    <visual name="left_eye">
      <origin xyz="0.15 -0.02 0.08" rpy="0 0 0"/>
      <geometry>
        <sphere  radius="0.01"/>
      </geometry>
      <material name="grey"/>
    </visual>
    <visual name="right_eye">
      <origin xyz="0.15 0.02 0.08" rpy="0 0 0"/>
      <geometry>
        <sphere  radius="0.01"/>
      </geometry>
      <material name="grey"/>
    </visual>
  </link>

  <link name="left_wheel">
    <collision name="collision">
      <origin xyz="0 0.0 0" rpy="0 1.5707 1.5707"/>
      <geometry>
         <cylinder  length="0.015" radius="0.02"/>
      </geometry>
    </collision>
    <visual name="visual">
      <origin xyz="0 0.0 0" rpy="0 1.5707 1.5707"/>
      <geometry>
        <cylinder  length="0.015" radius="0.02"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  
  <joint type="revolute" name="left_wheel_joint">
    <origin xyz="0 0.16 0.02" rpy="0 0 0"/>
    <child link="left_wheel">left_wheel</child>
    <parent link="base_link">base_link</parent>
    <axis xyz="0 1 0"/>
    <limit effort="100" velocity="100.0"  lower="-5000" upper="5000" />
  </joint> 

  <link name="right_wheel">
    <collision name="collision">
      <origin xyz="0 0.0 0" rpy="0 1.5707 1.5707"/>
      <geometry>
         <cylinder  length="0.015" radius="0.02"/>
      </geometry>
    </collision>
    <visual name="visual">
      <origin xyz="0 0.0 0" rpy="0 1.5707 1.5707"/>
      <geometry>
        <cylinder  length="0.015" radius="0.02"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  
  <joint type="revolute" name="right_wheel_joint">
    <origin xyz="0 -0.16 0.02" rpy="0 0 0"/>
    <child link="right_wheel">right_wheel</child>
    <parent link="base_link">base_link</parent>
    <axis xyz="0 1 0"/>
    <limit effort="100" velocity="100.0"  lower="-5000" upper="5000" />
  </joint>


  <link name="laser">
    <visual>
      <origin xyz="0 0.0 0" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.03"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>

  <joint type="fixed" name="laser_to_base_joint">
    <origin rpy="0 0 0" xyz="-0.05 0 0.05"/>
    <child link="laser"/>
    <parent link="base_link"/>
  </joint>

  <link name="imu">
    <visual>
      <origin xyz="0 0.0 0" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.01"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  
  <joint type="fixed" name="imu_to_base_joint">
    <origin rpy="0 0 -1.5708" xyz="0 0 0.06"/>
    <child link="imu"/>
    <parent link="base_link"/>
  </joint>

</robot>
